crypto/ethsecp256k1/ethsecp256k1.go

```diff
@@ -0,0 +1,249 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package ethsecp256k1
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/subtle"
+	"fmt"
+
+	errorsmod "cosmossdk.io/errors"
+	tmcrypto "github.com/cometbft/cometbft/crypto"
+	"github.com/cosmos/cosmos-sdk/codec"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	errortypes "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/evmos/os/ethereum/eip712"
+)
+
+const (
+	// PrivKeySize defines the size of the PrivKey bytes
+	PrivKeySize = 32
+	// PubKeySize defines the size of the PubKey bytes
+	PubKeySize = 33
+	// KeyType is the string constant for the Secp256k1 algorithm
+	KeyType = "eth_secp256k1"
+)
+
+// Amino encoding names
+const (
+	// PrivKeyName defines the amino encoding name for the EthSecp256k1 private key
+	PrivKeyName = "ethermint/PrivKeyEthSecp256k1"
+	// PubKeyName defines the amino encoding name for the EthSecp256k1 public key
+	PubKeyName = "ethermint/PubKeyEthSecp256k1"
```

Comment:
should these encoding names be changed to e.g. `os/PrivKeyEthSecp256k1` - would this require a migration for evmOS partners / the Evmos chain?

Reply:
@hanchon I'm not sure if this is going to cause some incompatibilities for any dApps or anything? 🤔 

```suggestion
	// PrivKeyName defines the amino encoding name for the EthSecp256k1 private key
	PrivKeyName = "os/PrivKeyEthSecp256k1"
	// PubKeyName defines the amino encoding name for the EthSecp256k1 public key
	PubKeyName = "os/PubKeyEthSecp256k1"
```